rainier-core/src/main/scala/com/stripe/rainier/core/SBC.scala

```diff
@@ -13,8 +13,8 @@ final case class SBC[T, L <: Distribution[T]](
 
   val priorGenerator = Generator.traverse(priorGenerators)
 
-  def posteriorSamples(nSamples: Int)(implicit rng: RNG): List[Double] =
-    posterior.map { case (_, r) => r }.record(nSamples).params.flatten
+  def posteriorSamples(nSamples: Int)(implicit rng: RNG): List[T] =
+    posterior.map { case (l, _) => l.generator }.sample(nSamples)
```

Comment:
This is what we had originally and we changed it so that it would always return a `List[Double]`. The problem is that that `List[Double]` consists of values of the prior parameters that we then feed into the model again in `replay` to get a sample from the posterior. For SBC, we actually want a sample from the posterior or else different posteriors with the same priors will have the same values here and checking goldsets won't really tell us much.

Reply:
Two things actually: `posterior.map { case (_, r) => r }.record(nSamples).params.flatten` is _definitely_ not what we want because `r` is literally just the prior again, but even `posterior.map { case (l, _) => l }.record(nSamples).params.flatten` is not what we want because it _also_ returns prior, this time because of how `record` works.